### DISCUSSION TITLE
_sif == 0 causing vsts to crash

### DISCUSSION THREAD

**Discussion by spamatica** [muse-sequencer/muse#1151] [Team Posts]
Sometimes I write tickets, sometimes I don't... it seems to be random..

Anyway. Helping a plugin dev (cut through recordings) to test some plugins and I found that we have started crashing on a few vst plugins.

The offending function is
void VstNativeEditor::open(MusECore::VstNativeSynthIF* sif, MusECore::VstNativePluginWrapper_State *state)

When opening the gui it is called with sif=0 and this causes a crash when referencing the new quirks code.
_sif->cquirks()._f ...

For testing I've just made a conditional to skip the quirk if it's 0 though probably the real solution is not that.

**Comment by terminator356** [comment]
Got this one. Looks like a simple oversight in kybos's fixNativeUIScaling quirk.
I know, the plugin wrapper thing is mind-bending, eh?

Added:
`const MusECore::PluginQuirks& quirks = _sif ? _sif->cquirks() : _pstate->pluginI->cquirks();'

And changed:
`_sif->cquirks()._fixNativeUIScaling`
 to 
`quirks._fixNativeUIScaling`

Please test, especially @kybos 
Thanks!

**Comment by kybos** [comment]
Thanks for the fix, I will test later.
Strange I did not catch it, I tested the code with all the VST plugins I have... So I don't expect any change in behaviour on my side.

**Comment by spamatica** [comment]
Thanks, working now =)
